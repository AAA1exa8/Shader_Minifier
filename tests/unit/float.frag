void main()
{
float f1 = 1.5;
float f2 = .0000003;
float f3 = .42;
float f4 = .0;
float f5 = 0.;
float f6 = -2e-3;
float f7 = 2E-9;
float f8 = 2E+6;
float f9 = 2e10;

float n = 1. / (f1 + f2 + f3 + f4 + f5);
float o = 1. / (f6 * f8 * f7 * f9);

gl_FragColor=vec4(n,o,n,0.);
}
