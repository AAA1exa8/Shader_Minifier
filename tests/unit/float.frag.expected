// Generated with  (https://github.com/laurentlb/Shader_Minifier/)
#ifndef SHADER_MINIFIER_IMPL
#ifndef SHADER_MINIFIER_HEADER
# define SHADER_MINIFIER_HEADER
#endif

#else // if SHADER_MINIFIER_IMPL

// tests/unit/float.frag
 "float floatPrecision()"
 "{"
   "float a=1.2345e-7,b=1.234567891234,c=1234567.891234,d=1.234567e14;"
   "return a*=b*=c*=d;"
 "}"
 "float largeNumbers()"
 "{"
   "float a=4.1e8,b=4.2e10,c=4.3e12,d=4.4e14;"
   "return a*=b*=c*=d;"
 "}"
 "float smallNumbers()"
 "{"
   "float a=.4,b=.04,c=.004,d=4e-4,e=4e-8,f=4e-10,g=4e-12,h=4e-14;"
   "return a*=b*=c*=d*=e*=f*=g*=h;"
 "}"
 "float zero()"
 "{"
   "float a=0.,b=0.,c=0.;"
   "return a*=b*=c;"
 "}"
 "void main()"
 "{"
   "float f1=1.5,f2=3e-7,f3=.42,f6=-.002,f7=2e-9,f8=2e6,f9=2e10;"
   "gl_FragColor=vec4(f1*=f2*=f3*=f6*=f7*=f8*=f9);"
 "}",

#endif
